lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -444,36 +444,30 @@ tokenize([H|_] = String, Line, Column, Scope, Tokens) when ?is_digit(H) ->
   {Rest, Number, Length} = tokenize_number(String, [], false),
   tokenize(Rest, Line, Column + Length, Scope, [{number, {Line, Column, Column + Length}, Number}|Tokens]);
 
-% Aliases
+% Identifiers (including aliases)
 
-tokenize([H|_] = Original, Line, Column, Scope, Tokens) when ?is_upcase(H) ->
-  {Rest, Alias, Length} = tokenize_identifier(Original, []),
-  case unsafe_to_atom(Alias, Line, Scope) of
-    {ok, Atom} ->
+tokenize([H|_] = Original, Line, Column, Scope, Tokens) when ?is_identifier_start(H) ->
+  case tokenize_any_identifier(Original, Line, Scope) of 
+    {ok, Rest, Atom, Length, HasAt, HasEnding} ->
       case Rest of
         [$:|T] when ?is_space(hd(T)) ->
           tokenize(T, Line, Column + Length + 1, Scope, [{kw_identifier, {Line, Column, Column + Length + 1}, Atom}|Tokens]);
+        [$:|T] when ?is_atom_start(hd(T)); ?is_digit(hd(T)) ->
```

Comment:
I think this could be `when hd(T) /= ?:`. We just need to check this clause in case we have `foo::bar`. Otherwise `foo:+` works and it is very weird. :)